File: Protogen/loxim_protocol/cpp/protocol/packages_data/Uint32Package.cpp
#include <string.h>
#include <stdlib.h>
#include "Uint32Package.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

Uint32Package::Uint32Package(\
		uint32_t a_value
		)
{
	value=a_value;
}

Uint32Package::Uint32Package()
{
	value=0;
}

Uint32Package::~Uint32Package()
{
}

void Uint32Package::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeUint32(value);
}

bool Uint32Package::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	value=reader->readUint32();
	return true;
}

bool Uint32Package::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	Uint32Package* p=(Uint32Package*)_package;
	if (!(getValue()==p->getValue())) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/Uint32Package.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class Uint32Package extends Package {
		public final static long ID=(long)5;
	
    	/**
**/
    	private long value=0;
    	
	public Uint32Package()
	{
	}

		public Uint32Package(
		    	long a_value    	)
	{
        	value=a_value;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			value=reader.readUint32();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeUint32(value);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		Uint32Package p=(Uint32Package)obj;
	    if (!((value==p.getValue()))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public long getValue(){return value;};
		
    	public void setValue(long a_value){ value=a_value;};
		
    
	 
}


